client/my-sites/stats/feedback/index.tsx

```diff
@@ -101,6 +104,10 @@ function StatsFeedbackController( { siteId }: FeedbackProps ) {
 		if ( action === FEEDBACK_ACTION_DISMISS_FLOATING_PANEL ) {
```

Comment:
I'll get this updated. Because the conditions are the same (`action === `), I think this makes sense as a `switch` actually. It clarifies the type of conditionals to expect.

I think different button handlers would make sense if the triggers where within this component. I think you're highlighting an issue with the logic though. It should probably be named differently to denote that it's a callback/action handler. In that way, we only need to pass the one function into child components but we clarify what we expect in the response.

Reply:
> I think this makes sense as a switch actually. It clarifies the type of conditionals to expect.

Yes I agree switch statement would be better if we have to make it one function.

> I think different button handlers would make sense if the triggers where within this component. I think you're highlighting an issue with the logic though. It should probably be named differently to denote that it's a callback/action handler. In that way, we only need to pass the one function into child components but we clarify what we expect in the response.

I don't quite follow here. If it's for different events to do different things, then we should separate them to different handlers just to separate concerns (Also the caller doesn't have to know all the legal event names for the function). I don't  see a few less click handlers passing to a component much beneficial. Secondly, passing handlers/parameters several levels downs is something we don't like, and that's the reason Redux is so popular. With functional components, we'll be able to share state with hooks, which would make it a lot clearer too. I guess we could come back and have a discussion what's the preferred approach to solve these problems in next maintenance sprint 🙂

Reply:
Follow up and fixes [here](https://github.com/Automattic/wp-calypso/pull/94129#discussion_r1770704572).